File: dataset/output-bk/GEMINI_gemini-1.5-pro-001/Task103_GEMINI_gemini-1.5-pro-001.java
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class PasswordChecker {

    public static void main(String[] args) {
        String password = "YourPasswordHere";
        checkPasswordStrength(password);
    }

    public static void checkPasswordStrength(String password) {
        if (password.length() < 8) {
            System.out.println("Password must be at least 8 characters long.");
            return;
        }

        Pattern pattern = Pattern.compile("(?=.*\\d)(?=.*[a-z])(?=.*[A-Z]).*");
        Matcher matcher = pattern.matcher(password);

        if (!matcher.matches()) {
            System.out.println("Password must contain at least one uppercase letter, one lowercase letter, and one digit.");
            return;
        }

        System.out.println("Strong password.");
    }
}

File: dataset/output/GEMINI_gemini-1.5-pro-001/Task103_GEMINI_gemini-1.5-pro-001.c
#include <stdio.h>
#include <string.h>
#include <ctype.h>

int main() {
    char password[100];
    int length = 0, uppercase = 0, lowercase = 0, digit = 0;

    printf("Enter password: ");
    scanf("%s", password);

    length = strlen(password);

    for (int i = 0; i < length; i++) {
        if (isupper(password[i])) {
            uppercase++;
        } else if (islower(password[i])) {
            lowercase++;
        } else if (isdigit(password[i])) {
            digit++;
        }
    }

    if (length >= 8 && uppercase > 0 && lowercase > 0 && digit > 0) {
        printf("Strong password.
");
    } else {
        printf("Weak password. ");
        if (length < 8) {
            printf("Password must be at least 8 characters long. ");
        }
        if (uppercase == 0) {
            printf("Password must contain at least one uppercase letter. ");
        }
        if (lowercase == 0) {
            printf("Password must contain at least one lowercase letter. ");
        }
        if (digit == 0) {
            printf("Password must contain at least one digit. ");
        }
        printf("
");
    }

    return 0;
}